libvita2d/source/vita2d_image_jpeg.c

```diff
@@ -6,23 +6,14 @@
 #include <jpeglib.h>
 #include "vita2d.h"
 
+// Max memory we want to use by jpg is 64 MB half of the VRAM, how every pixel is 4 bytes, then 64 * 1024 * 1024 / 4
+// Which is exactly the same than a picture of 4096 x 4096
+#define MAX_JPEG_MATRIX_SIZE 16777216
+
 static vita2d_texture *_vita2d_load_JPEG_generic(struct jpeg_decompress_struct *jinfo, struct jpeg_error_mgr *jerr)
 {
-	float downScaleWidth = (float)jinfo->image_width / 4096;
-	float downScaleHeight = (float)jinfo->image_height / 4096;
-	float downScale = (downScaleWidth >= downScaleHeight) ? downScaleWidth : downScaleHeight;
-
-	if (downScale <= 1.f) {
-		jinfo->scale_denom = 1;
-	} else if (downScale <= 2.f) {
-		jinfo->scale_denom = 2;
-	} else if (downScale <= 4.f) {
-		jinfo->scale_denom = 4;
-	} else if (downScale <= 8.f) {
-		jinfo->scale_denom = 8;
-	} else {
-		return NULL;
-	}
```

Comment:
The only cleanup that keeps the same functionality that comes to my mind here is just to replace the if-else with:
```c
jinfo->scale_denom = ceil(downScale);
```